src/protojson/WorkerCzarComIssue.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_PROTOJSON_WORKERCZARCOMISSUE_H
+#define LSST_QSERV_PROTOJSON_WORKERCZARCOMISSUE_H
+
+// System headers
+#include <chrono>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+
+// Third party headers
+#include "nlohmann/json.hpp"
+
+// qserv headers
+#include "protojson/WorkerQueryStatusData.h"
+
+// This header declarations
+namespace lsst::qserv::protojson {
+
+/// This class is used to send/receive a message from the worker to a specific
+/// czar when there has been a communication issue with the worker sending UberJob
+/// file ready messages. If there have been timeouts, the worker will send this
+/// message to the czar immediately after the worker receives a
+/// WorkerQueryStatusData message from the czar (indicating that communication
+/// is now possible).
+/// If communication with the czar has failed for a long time, the worker
+/// will set "_thoughtCzarWasDead" and delete all incomplete work associated
+/// with that czar. Result files will remain until garbage cleanup or the czar
+/// calls for their removal.
+/// TODO:UJ  UberJob complete messages that failed to be sent to the czar
+/// TODO:UJ  will be added to this message. (uberjob file response)
+/// Upon successful completion, the worker will clear all values set by the
+/// the czar.
+/// Currently, this message is expected to only be needed rarely.
+class WorkerCzarComIssue {
+public:
+    using Ptr = std::shared_ptr<WorkerCzarComIssue>;
+
+    WorkerCzarComIssue() = delete;
+    ~WorkerCzarComIssue() = default;
+
+    std::string cName(const char* funcN) { return std::string("WorkerCzarComIssue") + funcN; }
```

Comment:
I already commented a similar function in the context of the class `JobReadyMsg`.
Firstly, I would hide the implementation of the method in the CC file.
Secondly, the name cName is meaningless (to me)
Thirdly, the raw pointers are unsafe. I would at least check if the pointer is not `nullptr` and throw an exception to reinforce the application and avoid SIGFAILT should the null be found here.